src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -707,6 +724,29 @@ export class CellModel implements ICellModel {
 		return source;
 	}
 
+	// Run ADS commands via a notebook. Configured by a setting (turned off by default).
+	private checkForAdsCommandMagic(): boolean {
+		if (!this._isCommandExecutionSettingEnabled) {
+			return false;
+		}
+
+		let executeCommandMagic = '%%AZDATA_EXECUTE_COMMAND';
+		if (this._source && this._source.length) {
+			if (this._source[0].startsWith(executeCommandMagic)) {
```

Comment:
Is it valid to have spaces before a magic? And then to also have multiple spaces after the magic name but before the first parameter? Generally I feel like trimming in these cases is a good idea

Reply:
No, but I'll trim anyway.

Reply:
If that's actually not allowed for normal magics I'd probably say keep that behavior then - it'd just be confusing I think to have things behave differently based on what "type" of magic they are.